src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -63,13 +64,14 @@ export function TopNavMenu(props: TopNavMenuProps) {
   }
 
   function renderLayout() {
+    const className = props.noPadding ? undefined : 'kbnTopNavMenu';
```

Comment:
`kbnTopNavMenu` some top/bottom padding:
https://github.com/elastic/kibana/blob/daac96cce7a3844e03c93a4e1e0ad01b06ce69f6/src/plugins/navigation/public/top_nav_menu/_index.scss#L5

which with the new DOM structure was causing Dashboard in the fullscreen mode to have some empty space between the top of the screen and the first row of visualizations. We need to disable paddings in fullscreen mode.